### DISCUSSION TITLE
Is there a way to recover from conflicts?

### DISCUSSION THREAD

**Discussion by janniks** [gitext-rs/git-stack#342] [Q&A]
Is there a clean way to recover from conflicts?
_Similar to:_ https://github.com/gitext-rs/git-stack/discussions/225

In this case on `main` something was merged, which conflicted with a stack: `main←next←br1←br2`.

`git sync`
```sh
From github.com:...
 * branch              main       -> FETCH_HEAD
ERROR: Failed to re-stack branch `br2`: cherry-pick conflicts:
  ...
  ...
; class=Index (10); code=Unmerged (-10)
note: to undo, run `git branch-stash pop git-stack`
```

I was able to manually fix `next` with a simple `git rebase -i main` (from the current branch `next`). But this disconnects the branches `br1←br2` to unlabled commits, (annoying to filter out duplicated when rebasing later).

---

Is there a more elegant way to use `git-stack` to fix these conflicts?

> Also interesting, I have `git config --global --add --bool rebase.updateRefs true`, but they were still disconnected :(

**Comment by epage** [comment]
`git stack` tries to identify simple ways to remove conflicts (ie if `git rebase main` works, `git stack` is expected to work though bugs might exist).  The bigger problem is when manually intervention is needed, whether because you need to delete commits in `git rebase -i` or there are true conflicts.

For myself, I'm planning on learning `jj` as that handles this and many other cases well and hoping it can completely replace `git stack`.  We'll see.

**Reply by janniks** [reply]
Thanks for the response!

I really liked that git-stack doesn't need any new hidden folders etc.

Looking into jj, sapling, git-branchless as well. Graphite, ghstack, and spr were too opinionated, even though I like them.

Really good job on git-stack, it's awesome (when there's no conflicts).

---

I think it should be possible to get the repair flow (or sync or whatever) to interactive rebase when needed, and thus allow conflict resolution, while keeping the sync process intact (and never losing the branch hierarchy). But I'm no expert here, so idk
